behavior/MouseTracking/get_mouse_XY_pos.m

```diff
@@ -0,0 +1,75 @@
+function [ centroids ] = get_mouse_XY_pos( movie )
+% Extract the X,Y positions ( centroids ) of the mouse from the behavior
+%   video ( movie )
+%   
+% Input:
+%   movie: Behavior video (m4v); movie should be cropped (wavy curtain
+%   leads to fall centroids (haven't found another way to fix this)
+%
+% Returns matrix centroids where each row is an (x,y) coordinate of the
+%   mouse. 1st column = X, 2nd column = Y. Length of matrix =
+%   number of frames in movie
+%
+% Notes: Analyzes movie in chunks of 500 frames (smaller memory load)
+%   *** read function does not work properly on Linux, so unfortunately
+%   have to run this on a Windows machine ***
+%   *** Not very fast, but only have to run once; ~54K frames = ~40 min
+%
+% 2015-02-26 Fori Wang
+
+    behavior_vid = VideoReader(movie);
+    num_frames = behavior_vid.NumberofFrames;
+
+    % initialize centroids
+    centroids = cell(num_frames,1);
+    
+    % setup chunks of frames to read in movie
+    frame_indices = make_frame_indices(num_frames,1000);
+    
+    % setup background image: average of 5000 frames
+    bg_vid = read(behavior_vid,[1 5000]);
+    bg_image = mean(bg_vid,4);
+    bg_image = uint8(bg_image);
+    
+    for idx = 1:length(frame_indices)
+
+        fprintf('  Processing frames %d to %d (%s)\n',...
+            frame_indices(idx,1), frame_indices(idx,2), datestr(clock));
+
+        % read in all of the frames for the trial at the beginning
+        frame_range = [frame_indices(idx,1) frame_indices(idx,2)];
+        video = read(behavior_vid,frame_range);
+        
+        c_old = [0 0];
+
+        for frame_idx = frame_indices(idx,1)-(frame_indices(idx,1)-1):...
+                        frame_indices(idx,2)-(frame_indices(idx,1)-1)
+           
+           % Update original image CData
+            image = video(:,:,:,frame_idx);
```

Comment:
Doing so could speed up your code significantly, since it reduces all the frames from `[height width RGB]` to just `[height width]`. I see that `findMouse` has some nontrivial image operations in it.